src/components/Pagination/Pagination.tsx

```diff
@@ -1 +1,94 @@
-export const Pagination = () => {};
+import React, { useEffect } from 'react';
+import cn from 'classnames';
+
+interface Props {
+  total: number;
+  perPage: number;
+  currentPage: number;
+  onPageChange: (e: React.MouseEvent<HTMLAnchorElement>) => void;
+}
+
+const getArrayFromNum = (length: number, startValue: number = 0) => {
+  return Array.from({ length: length }, (_, i) => i + startValue + 1);
+};
+
+export const Pagination: React.FC<Props> = ({
+  total,
+  perPage,
+  currentPage,
+  onPageChange,
+}) => {
+  const pages = getArrayFromNum(Math.ceil(total / perPage));
+  const maxItemsPerPage =
+    pages.length === currentPage ? total % perPage : perPage;
```

Comment:
Ensure that `maxItemsPerPage` does not result in zero items when `total % perPage` is zero. Consider adjusting the logic to handle this case.